builtin/providers/aws/resource_aws_lambda_permission.go

```diff
@@ -189,7 +189,16 @@ func resourceAwsLambdaPermissionRead(d *schema.ResourceData, meta interface{}) e
 		statement, err = findLambdaPolicyStatementById(&policy, d.Id())
 		return resource.RetryableError(err)
 	})
+
 	if err != nil {
+		if tErr, isTimeout := err.(*resource.TimeoutError); isTimeout {
+			if awsErr, ok := tErr.LastError.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
+				// Policy really doesn't exist
+				d.SetId("")
```

Comment:
We usually also log `WARN` level log message here to make potential debugging easier - see [other resources](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_api_gateway_authorizer.go#L105) that have such logging.